Project setup

RikardFahlstrom opened the pull request.

RikardFahlstrom committed:
Add PyCharms directory to gitignore

RikardFahlstrom committed:
Add config file to gitignore

RikardFahlstrom committed:
Add basic info and run instructions to README

RikardFahlstrom committed:
Add first version of project files

victorsowa commented on browser_tools.py:
```diff
@@ -0,0 +1,105 @@
+        row_values = table_row.find_all('td')  # <td> is table cells
+        if row_values:
+            row_values = [row_value.text.strip() for row_value in row_values]
+            row_values = [y for x in row_values for y in x.split(':')]
```
Jag är öppen för att jag kanske bara är ovan med att läsa det. Men om man har två loopar i en list comprehension så kan det vara nice att köra det som en vanlig loop istället för läsbarhetens skull.

victorsowa commented on browser_tools.py:
```diff
@@ -0,0 +1,105 @@
+
+    for table_row in table_rows:
+        row_values = table_row.find_all('td')  # <td> is table cells
+        if row_values:
```
Om man kollar om det är None eller inte så ska man skriva ut det som `if row_values is not None`

RikardFahlstrom replied:
Fixat

victorsowa commented on browser_tools.py:
```diff
@@ -0,0 +1,105 @@
+    return soup
+
+
+def extract_and_transform_data(soup: BeautifulSoup) -> List[Dict]:
```
Kan det delas upp till flera funktioner? En som extraherar och en som transformerar.

RikardFahlstrom replied:
Ja det är nog bra, uppdaterat.

victorsowa commented on db_tools.py:
```diff
@@ -0,0 +1,94 @@
+    rekyl_errand_id = Column(BigInteger, nullable=False)
+    errand_status = Column(VARCHAR(length=50), nullable=False)
+    reporter = Column(VARCHAR(length=50), nullable=True, default='N/A')
+    apartment = Column(VARCHAR(length=50), nullable=False, default='N/A')
```
nullable false och default 'N/A' ?

RikardFahlstrom replied:
Tog bort nullable argumentet för alla kolumner som har default values.

victorsowa commented on db_tools.py:
```diff
@@ -0,0 +1,94 @@
+    session.close()
+
+
+def get_all_errands_statuses(db_session):
```
Den här returnerar inte alla errand statuses. Kanske ska heta `print_all_errands_statuses` istället.

RikardFahlstrom replied:
Uppdaterat.

victorsowa commented on db_tools.py:
```diff
@@ -0,0 +1,94 @@
+    session.close()
+
+
+def get_all_errands_since_date(db_session, start_date):
```
Samma här, jag tror att man vill att funktioner som heter get ska returnera något

RikardFahlstrom committed:
Refactor based on PR comments

- Refactor extract and transform function of
soup object into separate functions
- Remove nullable argument for db table creation
if default value exists
- Rename functions from get to print to clarify intent

RikardFahlstrom committed:
Update README with local run instructions

RikardFahlstrom committed:
Update README with correct run instructions

RikardFahlstrom committed:
Refactor

- Add browser option with local executable file
- Refactor from listcomp to function for easier reading
- Remove redundant data cleaning step for row_values

RikardFahlstrom committed:
Add sample config.ini file

RikardFahlstrom committed:
Fix all PyCharm warnings

RikardFahlstrom committed:
Add pre-commit rules to repo and requirements.txt

RikardFahlstrom committed:
Run pre-commit hooks against all files
